tests/test_snow_typer.py

```diff
@@ -0,0 +1,140 @@
+from functools import partial
+from unittest.mock import MagicMock
+
+import pytest
+import typer
+
+from snowflake.cli.api.commands.snow_typer import SnowTyper
+from snowflake.cli.api.output.types import MessageResult
+
+from typer.testing import CliRunner
+
+
+def app_factory(
+    pre_execute=None,
+    result_handler=None,
+    exception_handler=None,
+    post_execute=None,
+):
+    class _CustomTyper(SnowTyper):
+        @staticmethod
+        def pre_execute_callback():
+            if pre_execute:
```

Comment:
shouldn't all these implementations call `super`? isn't it the preferred way to use `SnowTyper` when adding own callbacks?

Currently you test callbacks invocations but there is no test confirming that callbacks already defined in `SnowTyper` work.

Reply:
Good point. I will add more tests for specific `SnowTyper` implementation.